src/dmd/cppmanglewin.d

```diff
@@ -1193,7 +1198,7 @@ private:
             rettype.accept(tmp);
             tmp.flags &= ~MANGLE_RETURN_TYPE;
         }
-        if (!type.parameters || !type.parameters.dim || noargs)
+        if (!type.parameters || !type.parameters.dim)
```

Comment:
All the code I looked had no args mangled into the destructor... and all my tests link. How did you manage to make it mangle with args?

Reply:
VC builds 2 dtors for class CC:
- one without arguments and return type: `??1CC@@UAE@XZ (public: virtual __thiscall CC::~CC(void))` 
- one with an integer argument `??_ECC@@UAEPAXI@Z (public: virtual void * __thiscall CC::`vector deleting destructor'(unsigned int))`. This is mangled as a regular function, but with a special name. This roughly is represented by the code
```
void* CC::~CC(int del)
{
   this->~CC();
   if (del)
      delete (char*)this;
   return (void*) this;
}
```
The latter is the one added to the vtable or called by an explicite `delete`. The fomer is the one called in the dtor chain. So both are necessary when mixing C++ and D in the class hierarchy.

Reply:
Okay great. Thanks for clarifying that! I didn't do enough testing.